test-suite/testkit/src/api.rs

```diff
@@ -360,6 +423,51 @@ where
 
         Err(error)
     }
+
+    fn response_to_api_result_new<R>(mut response: Response) -> api::ApiResult<R>
+    where
+        R: DeserializeOwned + 'static,
+    {
+        fn extract_error(response: &mut Response) -> Option<api::ApiError> {
+            let code = response.status();
+            let body = response.text().expect("Unable to get response text");
```

Comment:
However, if this will move into `ApiError::parse` you should be careful with `expect` / `unwrap` usage.